src/Mod/Draft/draftguitools/gui_edit.py

```diff
@@ -833,7 +729,8 @@ def evaluate_menu_action(self, labelname):
             actions = obj_gui_tools.evaluate_context_menu_action(self, obj, idx, action_label)
 
         elif action_label == "add point":
-            self.addPoint(self.event)
+            if obj and obj_gui_tools:
+                obj_gui_tools.addPoint(self.event)
```

Comment:
HELP! Sorry, I do not get how to fix this: should I `checkout `the `Draft: Edit, Split out addPoint in multiple methods in the GuiTools`, commit the new change and then squash it? but doing so it seems I create another "line" and I lose the possibility to include the last 4 commits with inteactive rebase... 
Also i'm trying to stay with GitKraken: I'm biased by my education towards graphical interfaces...

Reply:
If you do a "checkout" of an earlier commit, that changes your working copy to point to that commit, without a clear way to "replay" the later commits that you skipped over going backwards (you could use cherry-pick to get them again, but that's a lot of manual work). Also, if you checkout a specific commit (rather than a branch), you end up in "detached HEAD" state, where your working copy no longer points to a  branch, which is usually not what you want.

In this case, I would suggest using an interactive rebase. I'm not sure how the Git Kraken UI does this, but on the commandline you pass a "base" commit for the rebase, which in this case should be a commit in your current history *before* the one that you want to edit. It's ok if you use an older commit, typically the branching point of your own changes is fine. This is the commit that git will "rewind" to, so all commits *after* the selected commit can be edited.

It seems in kraken you can right click a commit (or maybe only branch, not entirely sure) and select "Interactive Rebase": https://support.gitkraken.com/working-with-repositories/interactive-rebase/

You should then get a list of commits that have been rewound and will be replayed, and you can select actions for each of them. The default is "pick" - just apply the commit and continue (internally, this is AFAIU exactly equivalent to cherry-pick). In this case, you would change the action for the commit you want to change to "Edit". After confirming, kraken should start applying the selected actions. When it comes to "edit", it applies that commit and then stops, so you can make any changes and commit them (with "amend" to modify the commit you want to edit, though you can also create a new commit to *insert* a new commit in the history). Once you are done, you "continue" the rebase, and git will proceed applying the rest of the selected commits.

Does that help?

Reply:
Indeed it helps! the point is that i think that gitKraken doesnt' provide the edit feature when interactive rebasing... I'll try with the command line...

Reply:
Heh, it seems you're right, it's not listed in the documentation either. I guess it's a bit harder for a GUI to offer this feature (on the comandline, you just drop into a shell and have to call `git rebase --continue` when you're done, for a GUI, it probably needs to keep track of the current state etc. OTOH, Kraken must surely implement this tracking anyway, in case you run into a conflict during a rebase, so it does surprise me a bit that they didn't implement the edit feature in Kraken.

In this case, I think you really need the rebase and edit feature. Often, an alternative is to commit a fixup commit at the end and then reorder and squash it, but in this case, you need to edit code that will be removed in a later commit, so you cannot create such a fixup commit at the end, since the code is already gone.

Note that if you get the rebase with edit going on the commandline, you might run into a conflict when continuing the rebase (since you'll have modified some code which is removed in a later commit, which likely produces a conflict when applying that later comit). Easy enough to fix, of course, just remove the code as intended :-)